src/features/game/expansion/components/resources/Iron.tsx

```diff
@@ -0,0 +1,339 @@
+import { useActor } from "@xstate/react";
+
+import { getTimeLeft } from "lib/utils/time";
+import { miningAudio, miningFallAudio } from "lib/utils/sfx";
+import { HealthBar } from "components/ui/HealthBar";
+import { LandExpansionRock } from "features/game/types/game";
+import {
+  canMine,
+  IRON_RECOVERY_TIME,
+  MINE_ERRORS,
+} from "features/game/events/landExpansion/ironMine";
+import { calculateBumpkinStamina } from "features/game/events/landExpansion/replenishStamina";
+import { TimeLeftOverlay } from "components/ui/TimeLeftOverlay";
+import { Overlay } from "react-bootstrap";
+import { Label } from "components/ui/Label";
+
+const POPOVER_TIME_MS = 1000;
+const HITS = 3;
+
+interface Props {
+  ironIndex: number;
+  expansionIndex: number;
+}
+
+export const Iron: React.FC<Props> = ({ ironIndex, expansionIndex }) => {
+  const { gameService, selectedItem } = useContext(Context);
+  const [game] = useActor(gameService);
+
+  const [showPopover, setShowPopover] = useState(true);
+  const [errorLabel, setErrorLabel] = useState<"noStamina" | "noPickaxe">();
+  const [popover, setPopover] = useState<JSX.Element | null>();
+
+  const [touchCount, setTouchCount] = useState(0);
+  // When to hide the wood that pops out
+  const [collecting, setCollecting] = useState(false);
+
+  const overlayRef = useRef<HTMLDivElement>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const sparkGif = useRef<SpriteSheetInstance>();
+  const minedGif = useRef<SpriteSheetInstance>();
+
+  const [showIronTimeLeft, setShowIronTimeLeft] = useState(false);
+
+  const { setToast } = useContext(ToastContext);
+  const expansion = game.context.state.expansions[expansionIndex];
+  const ironRock = expansion.iron?.[ironIndex] as LandExpansionRock;
+  const tool = "Stone Pickaxe";
+
+  const stamina = game.context.state.bumpkin
+    ? calculateBumpkinStamina({
+        nextReplenishedAt: Date.now(),
+        bumpkin: game.context.state.bumpkin,
+      })
+    : 0;
+
+  // Reset the shake count when clicking outside of the component
+  useEffect(() => {
+    const handleClickOutside = (event: any) => {
+      if (
+        containerRef.current &&
+        !containerRef.current.contains(event.target)
+      ) {
+        setTouchCount(0);
+      }
+    };
+    document.addEventListener("click", handleClickOutside, true);
+    return () => {
+      document.removeEventListener("click", handleClickOutside, true);
+    };
+  }, []);
+
+  // Users will need to refresh to strike the iron again
+  const mined = !canMine(ironRock);
+
+  const displayPopover = async (element: JSX.Element) => {
+    setPopover(element);
+    setShowPopover(true);
+
+    await new Promise((resolve) => setTimeout(resolve, POPOVER_TIME_MS));
+    setShowPopover(false);
+  };
+
+  const hasPickaxes =
+    selectedItem === tool && game.context.state.inventory[tool]?.gte(1);
+  const hasStamina = stamina >= IRON_MINE_STAMINA_COST;
+
+  const strike = () => {
+    if (mined) {
+      return;
```

Comment:
This could be an inline `return` as well to be consistent with the returns below